server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java

```diff
@@ -394,12 +401,19 @@ public static RealmRepresentation toRepresentation(RealmModel realm, boolean int
         rep.setWebAuthnPolicyPasswordlessAvoidSameAuthenticatorRegister(webAuthnPolicy.isAvoidSameAuthenticatorRegister());
         rep.setWebAuthnPolicyPasswordlessAcceptableAaguids(webAuthnPolicy.getAcceptableAaguids());
 
+        CIBAPolicy cibaPolicy = realm.getCIBAPolicy();
```

Comment:
@pedroigor Does it mean that we do not use `CIBAPolicy` but instead implement getter method for each items that consist of the current `CIBAPolicy` directly onto `RealmModel` ?

```
        rep.setCibaBackchannelTokenDeliveryMode(realm.getBackchannelTokenDeliveryMode());
        rep.setCibaExpiresIn(realm.getExpiresIn());
        rep.setCibaInterval(realm.getInterval());
        rep.setCibaAuthRequestedUserHint(realm.getAuthRequestedUserHint());
```

 If so, I agree on this idea.

Reply:
What you said is basically how we do things today. I was thinking about the opposite, where CIBA related properties are grouped within a top-level type as you did for the `RealmModel`.

It sounds like a nitpick, but we have been discussing how to improve our extensibility so that you don't need to change core components such as the representation or model, when a new feature like CIBA needs to be introduced.

For this concrete case, I think having `CIBAPolicy` in the representation (similarly as you did to the model) helps to track (at least for now, considering we are not yet there in terms of our extensibility goals) CIBA related code more easily. And a possible smother refactoring in the future.

Reply:
@pedroigor Thank you for the clarification. I would like to follow your idea, namely managing CIBA related properties without changing core components like representations and models.